java-extras/src/main/java/org/triplea/java/collections/CollectionUtils.java

```diff
@@ -113,4 +115,15 @@
     return Iterables.elementsEqual(c1, c2)
         || (c1.size() == c2.size() && c2.containsAll(c1) && c1.containsAll(c2));
   }
+
+  /**
+   * Creates a sorted, mutable collection containing the specified elements that will maintain its
```

Comment:
Wow, this is quite the method :+1: 

Looks to be a very good use of priority queue, I like particularly how the data structure can just "take care of itself" and does the right thing.

Reply:
I want to point out that calling a `PriorityQueue` "sorted" is misleading.
When using it like this:
```java
final var queue = new PriorityQueue<>(elements.size(), comparator);
sortedCollection.addAll(elements);
while (!sortedCollection.isEmpty()) {
  final var element = sortedCollection.remove();
// Elements are returned in correct order
}
```
it works as expected, but the [JavaDocs explicitly state](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/PriorityQueue.html) that:
> The Iterator provided in method iterator() and the Spliterator provided in method spliterator() are not guaranteed to traverse the elements of the priority queue in any particular order.

With that being said, a TreeSet might be the better data structure here, but only if the keys are unique so far (it doesn't permit multiple elements that the comparator considers "equal", even if equals returns false)

Reply:
Apparently there's a [`TreeMultiset`](https://guava.dev/releases/21.0/api/docs/com/google/common/collect/TreeMultiset.html) in guava, that should work too

Reply:
Thanks! Indeed I missed that PriorityQueue iteration order doesn't do what we want. Switched to TreeMultiset and added some unit tests. :)

Reply:
Yeah, I ran into this problem as well when I tried messing around with it for Tile rendering code, that's why I immediately noticed ^^